server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionManager.java

```diff
@@ -78,10 +81,82 @@ private static long getDefaultThroughput(AccumuloConfiguration aconf) {
           .getMemoryAsBytes(Property.TSERV_COMPACTION_SERVICE_DEFAULT_RATE_LIMIT.getDefaultValue());
     }
 
-    Config(AccumuloConfiguration aconf) {
+    @SuppressWarnings("removal")
+    private Map<String,String> getConfiguration(AccumuloConfiguration aconf) {
+
       Map<String,String> configs =
           aconf.getAllPropertiesWithPrefix(Property.TSERV_COMPACTION_SERVICE_PREFIX);
 
+      // check if deprecated properties for compaction executor are set
+      if (aconf.isPropertySet(Property.TSERV_MAJC_MAXCONCURRENT, true)) {
+
+        String defaultServicePrefix =
+            Property.TSERV_COMPACTION_SERVICE_PREFIX.getKey() + DEFAULT_SERVICE.canonical() + ".";
+        boolean defaultServicePropsSet = false;
+
+        // check if any properties for the default compaction service are set
+        for (String key : configs.keySet()) {
+          if (key.startsWith(defaultServicePrefix)) {
+            Property prop = Property.getPropertyByKey(key);
+            if (prop == null || aconf.isPropertySet(prop, true)) {
+              defaultServicePropsSet = true;
+              break;
+            }
+          }
+        }
+
+        if (defaultServicePropsSet) {
+
+          String warning = String.format(
+              "The deprecated property %s was set. Properties with the prefix %s "
+                  + "were also set, which replace the deprecated properties. The deprecated "
+                  + "property was therefore ignored.",
+              Property.TSERV_MAJC_MAXCONCURRENT.getKey(), defaultServicePrefix);
+
+          if (!warning.equals(lastDeprecationWarning)) {
+            log.warn(warning);
+            lastDeprecationWarning = warning;
+          }
+        } else {
+          String numThreads = aconf.get(Property.TSERV_MAJC_MAXCONCURRENT);
+
+          // Its possible a user has configured the others compaction services, but not the default
+          // service. In this case want to produce a config with the default service configs
+          // overridden using deprecated configs.
+
+          HashMap<String,String> configsCopy = new HashMap<>(configs);
+
+          Map<String,String> defaultServiceConfigs =
+              Map.of(defaultServicePrefix + "planner", DefaultCompactionPlanner.class.getName(),
+                  defaultServicePrefix + "planner.opts.executors",
+                  "[{'name':'deprecated','numThreads':" + numThreads + "}]");
+
+          configsCopy.putAll(defaultServiceConfigs);
+
+          String warning = String.format(
+              "The deprecated property %s was set.  Properties with the prefix %s "
```

Comment:
You have some inconsistency, sometimes using double spaces after a period, and sometimes using a space after a period. Although there is debate over which is best, I recommend standardizing on a single space after the period. It is more concise, and easier to predict and grep for in the logs. When they show on the monitor page and presented as HTML, they get collapsed to a single space anyway, so matching in a browser with Ctrl-F is also consistent with grepping for a single space in the logs.